internal/provider/resource_instance_test.go

```diff
@@ -1095,12 +1097,292 @@ resource "oxide_instance" "{{.BlockName}}" {
+								plancheck.ExpectResourceAction(resourceName, plancheck.ResourceActionReplace),
+							},
+						},
+						Check: resource.ComposeAggregateTestCheckFunc([]resource.TestCheckFunc{
+							resource.TestCheckResourceAttr(resourceName, tc.from, "terraform-acc-myhost-updated"),
+							resource.TestCheckResourceAttr(resourceName, tc.to, "terraform-acc-myhost-updated"),
+						}...),
+					},
+					// Update hostname attribute name and value.
+					// Expect a resource replacement.
+					{
+						Config: generateConfig(t, instanceName, map[string]string{tc.to: "terraform-acc-myhost"}),
+						ConfigPlanChecks: resource.ConfigPlanChecks{
+							PreApply: []plancheck.PlanCheck{
+								plancheck.ExpectResourceAction(resourceName, plancheck.ResourceActionReplace),
+							},
+						},
+						Check: resource.ComposeAggregateTestCheckFunc([]resource.TestCheckFunc{
+							resource.TestCheckResourceAttr(resourceName, tc.to, "terraform-acc-myhost"),
+							resource.TestCheckResourceAttr(resourceName, tc.from, "terraform-acc-myhost"),
+						}...),
+					},
+				},
+			})
+		})
+	}
+
+	// Test resource import.
+	t.Run("import with hostname", func(t *testing.T) {
+		instanceName := newResourceName()
+
+		resource.ParallelTest(t, resource.TestCase{
+			PreCheck:                 func() { testAccPreCheck(t) },
+			ProtoV6ProviderFactories: testAccProtoV6ProviderFactories(),
+			CheckDestroy:             testAccInstanceDestroy,
+			Steps: []resource.TestStep{
+				{
+					Config: generateConfig(t, instanceName, map[string]string{
+						"hostname": "terraform-acc-myhost",
+					}),
+					Check: resource.ComposeAggregateTestCheckFunc([]resource.TestCheckFunc{
+						resource.TestCheckResourceAttr(resourceName, "hostname", "terraform-acc-myhost"),
+						resource.TestCheckResourceAttr(resourceName, "host_name", "terraform-acc-myhost"),
+					}...),
+				},
+				{
+					ImportState:             true,
+					ResourceName:            resourceName,
+					ImportStateVerify:       true,
+					ImportStateVerifyIgnore: []string{"start_on_create"},
+				},
+			},
+		})
+	})
+
+	t.Run("import with host_name", func(t *testing.T) {
+		instanceName := newResourceName()
+
+		resource.ParallelTest(t, resource.TestCase{
+			PreCheck:                 func() { testAccPreCheck(t) },
+			ProtoV6ProviderFactories: testAccProtoV6ProviderFactories(),
+			CheckDestroy:             testAccInstanceDestroy,
+			Steps: []resource.TestStep{
+				{
+					Config: generateConfig(t, instanceName, map[string]string{
+						"host_name": "terraform-acc-myhost",
+					}),
+					Check: resource.ComposeAggregateTestCheckFunc([]resource.TestCheckFunc{
+						resource.TestCheckResourceAttr(resourceName, "host_name", "terraform-acc-myhost"),
+						resource.TestCheckResourceAttr(resourceName, "hostname", "terraform-acc-myhost"),
+					}...),
+				},
+				{
+					ImportState:        true,
+					ResourceName:       resourceName,
+					ExpectNonEmptyPlan: true,
```

Comment:
```suggestion
					ImportState:             true,
					ResourceName:            resourceName,
					ImportStateVerify:       true,
					ImportStateVerifyIgnore: []string{"start_on_create"},
```

Ops, I think I forgot to update this part of the test. Without `ImportStateVerify` this step is not actually testing much.